GAE/src/org/waterforpeople/mapping/app/web/CronCommanderServlet.java

```diff
@@ -212,4 +231,172 @@ private void purgeOrphanJobQueueRecords() {
             }
         }
     }
+
+    /**
+     * scans for and extracts geotags from image answers less than 1 month old
+     * Intended to be run every day
+     */
+    private void extractImageFileGeotags() {
+        Calendar deadline = Calendar.getInstance();
+        deadline.add(Calendar.MONTH, ONE_MONTH_AGO);
+        log.info("Starting scan for image answers, newer than: " + deadline.getTime());
+        QuestionAnswerStoreDao qaDao = new QuestionAnswerStoreDao();
+        Long cursor = null;
+        List<QuestionAnswerStore> dfjqList;
+        int retirees = 0;
+        int json = 0;
+        int nonjson = 0;
+        Media media;
+
+        do {
+            dfjqList = qaDao.listByTypeAndDate("IMAGE", cursor, deadline.getTime(), null, 1000);
+            if (dfjqList.size() == 0) break; //no more answers
+
+            //loop over this batch
+            for (QuestionAnswerStore item : dfjqList) {
+                boolean forceSave = false;
+                String v = item.getValue();
+                log.fine(String.format(" Old IMAGE value '%s'", v));
+                if (v != null && !v.trim().equals("")) {
+                    if (v.startsWith("{")) {
+                        json++;
+                        //Parse it
+                        media = MediaResponse.parse(v);
+                        if (media.getLocation() != null) { //Best case: Already known (could check validity)
+                            //System.out.println("Location in " + q);
```

Comment:
I would maybe remove this commented out line

Reply:
Done